odoo/addons/base/models/ir_qweb.py

```diff
@@ -148,31 +722,880 @@ def is_child_view(view_name):
-        return super()._compile_directive(el, options, directive, indent)
+        compile_handler = getattr(self, f"_compile_directive_{directive.replace('-', '_')}", None)
+        return compile_handler(el, options, indent)
+
+    # compile directives
+
+    def _compile_directive_debug(self, el, options, indent):
+        """Compile `t-debug` expressions into a python code as a list of
+        strings.
+
+        The code will contains the call to the debugger chosen from the valid
+        list.
+        """
+        debugger = el.attrib.pop('t-debug')
+        code = []
+        if options.get('dev_mode'):
+            code.append(self._indent(f"self._debug_trace({repr(debugger)}, compile_options)", indent))
+        else:
+            _logger.warning("@t-debug in template is only available in qweb dev mode options")
+        return code
+
+    def _compile_directive_options(self, el, options, indent):
+        """
+        compile t-options and add to the dict the t-options-xxx `t_options`
+        dictionary in compiled code.
+        """
+        code = []
+        dict_arg = []
+        for key in list(el.attrib):
+            if key.startswith('t-options-'):
+                value = el.attrib.pop(key)
+                option_name = key[10:]
+                dict_arg.append(f'{repr(option_name)}:{self._compile_expr(value)}')
+
+        t_options = el.attrib.pop('t-options', None)
+        if t_options and dict_arg:
+            code.append(self._indent(f"t_options = {{**{self._compile_expr(t_options)}, {', '.join(dict_arg)}}}", indent))
+        elif dict_arg:
+            code.append(self._indent(f"t_options = {{{', '.join(dict_arg)}}}", indent))
+        elif t_options:
+            code.append(self._indent(f"t_options = {self._compile_expr(t_options)}", indent))
+
+        el.set('t-consumed-options', str(bool(code)))
+
+        if not code:
+            return [self._indent("t_options = {}", indent)]
+
+        return code
+
+    def _compile_directive_consumed_options(self, el, options, indent):
+        raise ValueError('the t-options must be on the same tag as a directive that consumes it (for example: t-out, t-field, t-call)')
+
+    def _compile_directive_att(self, el, options, indent):
+        """ Compile the attributes of the given elements.
+
+        The compiled function will create the ``attrs`` dictionary. Then the
+        ``attrs`` dictionary will be output.
+
+        The new namespaces of the current element.
+
+        The static attributes (not prefixed by ``t-``) are add to the
+        ``attrs`` dictionary in first.
+
+        The dynamic attributes values will be add after. The dynamic
+        attributes has different origins.
+        - value from key equal to ``t-att``: python dictionary expression;
+        - value from keys that start with ``t-att-``: python expression;
+        - value from keys that start with ``t-attf-``: format string
+            expression.
+        """
+        code = []
```

Comment:
`code` always as a first element, it will be set below
```suggestion
        code = [None]
```

Reply:
Why ? If we add strings and then make a joint it will make an error.